src/java.base/share/classes/jdk/internal/foreign/layout/ValueLayouts.java

```diff
@@ -204,268 +199,240 @@ final V self() {
-            return new OfFloatImpl(order(), bitAlignment, name);
-        }
-
-        @Override
-        public OfFloatImpl withOrder(ByteOrder order) {
-            Objects.requireNonNull(order);
-            return new OfFloatImpl(order, bitAlignment(), name());
+        OfFloatImpl dup(ByteOrder order, long bitAlignment, Optional<String> name) {
+            return new OfFloatImpl(order, bitAlignment, name);
         }
 
         public static OfFloat of(ByteOrder order) {
-            return new OfFloatImpl(order);
+            return new OfFloatImpl(order, Float.SIZE, Optional.empty());
         }
     }
 
     public static final class OfLongImpl extends AbstractValueLayout<OfLongImpl> implements ValueLayout.OfLong {
 
-        private OfLongImpl(ByteOrder order) {
-            super(long.class, order, 64);
-        }
-
         private OfLongImpl(ByteOrder order, long bitAlignment, Optional<String> name) {
-            super(long.class, order, 64, bitAlignment, name);
-        }
-
-        @Override
-        OfLongImpl dup(long bitAlignment, Optional<String> name) {
-            return new OfLongImpl(order(), bitAlignment, name);
+            super(long.class, order, Long.SIZE, bitAlignment, name);
         }
 
         @Override
-        public OfLongImpl withOrder(ByteOrder order) {
-            Objects.requireNonNull(order);
-            return new OfLongImpl(order, bitAlignment(), name());
+        OfLongImpl dup(ByteOrder order, long bitAlignment, Optional<String> name) {
+            return new OfLongImpl(order, bitAlignment, name);
         }
 
         public static OfLong of(ByteOrder order) {
-            return new OfLongImpl(order);
+            return new OfLongImpl(order, Long.SIZE, Optional.empty());
         }
     }
 
     public static final class OfDoubleImpl extends AbstractValueLayout<OfDoubleImpl> implements ValueLayout.OfDouble {
 
-        private OfDoubleImpl(ByteOrder order) {
-            super(double.class, order, 64);
-        }
-
         private OfDoubleImpl(ByteOrder order, long bitAlignment, Optional<String> name) {
-            super(double.class, order, 64, bitAlignment, name);
+            super(double.class, order, Double.SIZE, bitAlignment, name);
         }
 
         @Override
-        OfDoubleImpl dup(long bitAlignment, Optional<String> name) {
-            return new OfDoubleImpl(order(), bitAlignment, name);
-        }
-
-        @Override
-        public OfDoubleImpl withOrder(ByteOrder order) {
-            Objects.requireNonNull(order);
-            return new OfDoubleImpl(order, bitAlignment(), name());
+        OfDoubleImpl dup(ByteOrder order, long bitAlignment, Optional<String> name) {
+            return new OfDoubleImpl(order, bitAlignment, name);
         }
 
         public static OfDouble of(ByteOrder order) {
-            return new OfDoubleImpl(order);
+            return new OfDoubleImpl(order, isAixOS ? 32 : Double.SIZE, Optional.empty());
```

Comment:
Please remove `isAixOS` and use `Utils.IS_AIX` instead.

Reply:
Fixed as suggested above.